server/proxy/pf_channels.c

```diff
@@ -59,6 +61,26 @@ void pf_OnChannelConnectedEventHandler(void* context,
 		server = ps->gfx;
 		pf_rdpgfx_pipeline_init(gfx, server, pc->pdata);
 	}
+	else if (strcmp(e->name, DISP_DVC_CHANNEL_NAME) == 0)
+	{
+		UINT error;
+		pc->disp = (DispClientContext*) e->pInterface;
+
+		if ((error = ps->disp->Open(ps->disp)) != CHANNEL_RC_OK)
+		{
+			if (error == ERROR_NOT_FOUND)
+			{
+				/* disp is not opened by client, ignore */
+				return;
+			}
+
+			WLog_WARN(TAG, "Failed to open disp channel");
+			return;
+		}
+
+		ps->dispOpened = TRUE;
+		pf_disp_pipeline_init(pc->disp, ps->disp, pc->pdata);
```

Comment:
there's no pipeline for disp, should better call it `pf_edisp_init`

Reply:
changed to pf_disp_register_callbacks